ipaserver/plugins/config.py

```diff
@@ -266,6 +277,103 @@ def show_servroles_attributes(self, entry_attrs, **options):
             config = backend.config_retrieve(role)
             entry_attrs.update(config)
 
+    def _gather_trusted_domains(self):
+        """
+        Aggregate all trusted domains into a dict keyed by domain names with
+        values corresponding to domain status (enabled/disabled)
+        """
+        command = self.api.Command
+        try:
+            ad_forests = command.trust_find()['result']
+        except errors.NotFound:
+            return {}
+
+        trusted_domains = {}
+        for forest_name in [a['cn'][0] for a in ad_forests]:
+            forest_domains = command.trustdomain_find(forest_name)['result']
+
+            trusted_domains.update(
+                {
+                    dom['cn'][0]: dom['domain_enabled'][0]
+                    for dom in forest_domains if 'domain_enabled' in dom
+                }
+            )
+
+        return trusted_domains
```

Comment:
It would be nice to share code with `check_associateddomain_is_trusted`. Not a blocker.